src/pretix/control/forms/filter.py

```diff
@@ -811,6 +811,211 @@ def meta_properties(self):
+
+        self.fields['provider'].choices += get_all_payment_providers()
+
+    def filter_qs(self, qs):
+        fdata = self.cleaned_data
+
+        if fdata.get('created_from'):
+            date_start = make_aware(datetime.combine(
+                fdata.get('created_from'),
+                time(hour=0, minute=0, second=0, microsecond=0)
+            ), get_current_timezone())
+            qs = qs.filter(created__gte=date_start)
+
+        if fdata.get('created_until'):
+            date_end = make_aware(datetime.combine(
+                fdata.get('created_until') + timedelta(days=1),
+                time(hour=0, minute=0, second=0, microsecond=0)
+            ), get_current_timezone())
+            qs = qs.filter(created__lt=date_end)
+
+        if fdata.get('completed_from'):
+            date_start = make_aware(datetime.combine(
+                fdata.get('completed_from'),
+                time(hour=0, minute=0, second=0, microsecond=0)
+            ), get_current_timezone())
+            qs = qs.filter(payment_date__gte=date_start)
+
+        if fdata.get('completed_until'):
+            date_end = make_aware(datetime.combine(
+                fdata.get('completed_until') + timedelta(days=1),
+                time(hour=0, minute=0, second=0, microsecond=0)
+            ), get_current_timezone())
+            qs = qs.filter(payment_date__lt=date_end)
+
+        if fdata.get('event'):
+            qs = qs.filter(order__event=fdata.get('event'))
+
+        if fdata.get('organizer'):
+            qs = qs.filter(order__event__organizer=fdata.get('organizer'))
+
+        if fdata.get('state'):
+            qs = qs.filter(state=fdata.get('state'))
+
+        if fdata.get('provider'):
+            qs = qs.filter(provider=fdata.get('provider'))
+
+        if fdata.get('query'):
+            u = fdata.get('query')
+
+            matching_invoices = Invoice.objects.filter(
+                Q(invoice_no__iexact=u)
+                | Q(invoice_no__iexact=u.zfill(5))
+                | Q(full_invoice_no__iexact=u)
+            ).values_list('order_id', flat=True)
+
+            matching_invoice_addresses = InvoiceAddress.objects.filter(
+                Q(
+                    Q(name_cached__icontains=u) | Q(company__icontains=u)
+                )
+            ).values_list('order_id', flat=True)
+
+            if "-" in u:
+                code = (Q(order__event__slug__icontains=u.rsplit("-", 1)[0])
+                        & Q(order__code__icontains=Order.normalize_code(u.rsplit("-", 1)[1])))
+            else:
+                code = Q(order__code__icontains=Order.normalize_code(u))
+
+            matching_orders = OrderPayment.objects.filter(
+                Q(
+                    code
+                    | Q(order__email__icontains=u)
+                    | Q(order__comment__icontains=u)
+                )
+            ).values_list('id', flat=True)
```

Comment:
This returns a list of OrderPayment IDs that is later treated like a list of Order IDs, so it will yield wrong results.

```suggestion
            matching_orders = Order.objects.filter(
                Q(
                    code
                    | Q(email__icontains=u)
                    | Q(comment__icontains=u)
                )
            ).values_list('id', flat=True)
```

The ``code``  definition will need to be adjusted likewise